### PR TITLE
Updates the development guide

### PR DESCRIPTION
**Pull request by josh146**
**Context:** The development guide and 'writing the docs' pages were out of date with the latest documentation refactor and PennyLane release.

**Description of the Change:**

* Updates the development guide

* Adds instructions for installing/developing the interfaces

* Adds a section on submitting a pull request

* Removes the page on 'writing the docs', and instead folds it in as a section in the development guide.

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** #362

### CODE REVIEW

File: doc/development/guide.rst [hunk 1]
```diff
@@ -1,79 +1,233 @@
...
+automatically to the comment box when you create a new issue.
+
+* When describing the pull request, please include as much detail as possible
+  regarding the changes made/new features added/performance improvements. If including any
+  bug fixes, mention the issue numbers associated with the bugs.
+
+* Once you have submitted the pull request, three things will automatically occur:
+
+  - The **test suite** will automatically run on `Travis CI <https://travis-ci.org/XanaduAI/pennylane>`_
+    to ensure that the all tests continue to pass.
```
Comment (antalszava) [inline doc/development/guide.rst]: "the" is not needed here/ comes as "all the"

Comment (josh146) [reply doc/development/guide.rst]: Good catch!

File: doc/development/guide.rst [hunk 2]
```diff
@@ -1,79 +1,233 @@
...
-Alternatively, you can install PennyLane from the source code on Github by navigating to an appropriate installation directory on your system and running
-::
 
-	$ git clone https://github.com/XanaduAI/pennylane
-        $ cd pennylane
-        $ pip install -e .
+.. note::
 
+    Due to the use of :ref:`entry points <installing_plugin>` to install
+    plugins means that changes to PennyLane device class locations or shortnames
```
Comment (mariaschuld) [inline doc/development/guide.rst]:
```suggestion
    plugins, changes to PennyLane device class locations or shortnames
```

File: doc/development/guide.rst (hunk 1 above)
Comment (josh146) [reply doc/development/guide.rst]:
```suggestion
    to ensure that all tests continue to pass.
```

File: doc/development/guide.rst [hunk 3]
```diff
@@ -1,79 +1,233 @@
...
 
 PennyLane requires the following libraries be installed:
 
 * `Python <http://python.org/>`_ >=3.5
 
 as well as the following Python packages:
 
-* `numpy <http://numpy.org/>`_  >=1.13.3
-* `scipy <http://scipy.org/>`_  >=1.0.0
+* `numpy <http://numpy.org/>`_ >=1.13.3
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/development/guide.rst]: Not that I know of :thinking:

File: doc/development/guide.rst [hunk 4]
```diff
@@ -1,79 +1,233 @@
...
-::
 
-	$ git clone https://github.com/XanaduAI/pennylane
-        $ cd pennylane
-        $ pip install -e .
+.. note::
 
+    Due to the use of :ref:`entry points <installing_plugin>` to install
+    plugins, changes to PennyLane device class locations or shortnames
+    requires ``pip install -e .`` to be re-run for the changes to take effect.
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/development/guide.rst]: Depends; the repo where the plugin is. So if it's default.qubit, then PennyLane core repo, if an external plugin, then the external plugin repo. Will clarify

File: doc/development/guide.rst [hunk 5]
```diff
@@ -1,79 +1,233 @@
...
+.. code-block:: bash
+
+    make docs
 
 The documentation can then be found in the :file:`doc/_build/html/` directory.
+
+.. note::
+
+    To build the interfaces documentation, PyTorch and TensorFlow will need to
+    be installed, see :ref:`install_interfaces`.
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/development/guide.rst]: Will try and fix again in a new PR

**Pull request merged**